src/components/Postiterie.vue

```diff
@@ -1,203 +1,306 @@
+      IN : x, y (numbers) coordinates from which to start croping
+           width, height (numbers) length to crop
+      OUT: imageData
+    */
+    crop: function( x, y, width, height){
+      let canvas = document.createElement('canvas');
+      let context = canvas.getContext('2d');
+      let img = document.getElementById('preview');
+      canvas.width = img.width;
+      canvas.height = img.height;
+      context.drawImage(img, 0, 0);
+      return context.getImageData(x, y, width, height);
+    },
+
     /*
-      FUNCTION previewImage
-      preview the uploaded image (and activate the button to track)
+      FUNCTION getImage
+      get the uploaded image and start the tracking
       IN : fieldName : the fieldName ;)
            file : the image
     */
-    previewImage: function(fieldName, file) {
-            this.isNoImage = false;
-            let imageFile = file[0] 
-            let formData = new FormData()
-            let imageURL = URL.createObjectURL(imageFile)
-            formData.append(fieldName, imageFile)
+    getImage: function(fieldName, file) {
+            this.link = null;
+            let imageFile = file[0];
+            let formData = new FormData();
+            let imageURL = URL.createObjectURL(imageFile);
+            formData.append(fieldName, imageFile);
             this.image=imageURL;
     },
 
     /*
       FUNCTION tracking
-      Track post-its and draw a square around them
+      Track post-its and add them to results[]
+      Then, send the file
+      TODO : investigate odd behavior --> the function doesn't work if not started from a onclick event (likely due to tracker.on('track', function(event))) )
     */
     tracking: function () {
       
       let $this = this;
-      let img = document.getElementById('preview');
-      let demoContainer = document.querySelector('.container');
+      $this.results = [];
+
       let tracker = new tracking.ColorTracker(['magenta', 'cyan', 'yellow']);
 
       tracker.on('track', function(event) {
-        event.data.forEach(function(rect) {
-          window.plot(rect.x, rect.y, rect.width, rect.height, rect.color);
-          let croppedImageData = crop(rect.x,rect.y,rect.width,rect.height);
-          let croppedImage = imageDataToImage(croppedImageData);
-          $this.results.push(croppedImage);
+
+        let images = event.data;
+        const croppedImagePromises = images.map((image)=>{
+          return new Promise(function(resolve){
+            let croppedImageData = $this.crop(image.x,image.y,image.width,image.height);
+            let croppedImage = $this.imageDataToImage(croppedImageData);
+            $this.results.push(croppedImage);
+            resolve();
+          });
         });
-      });
 
-      this.isNoResult = false;
+        Promise.all(croppedImagePromises).then(()=>{
+          console.log($this.results);
+          if ($this.results.length){
+              if ($this.authorized){
+                $this.sendFile();
```

Comment:
This can be simplified even more.

`.then()` calls can be chained and will be called in order when the previous promise resolves.